pkg/loader/load_test.go

```diff
@@ -378,6 +378,141 @@ func (s *batchManagerSuite) TestShouldExecAccumulatedDMLs(c *check.C) {
 	c.Assert(bm.txns, check.HasLen, 1)
 }
 
+type txnManagerSuite struct{}
+
+var _ = check.Suite(&txnManagerSuite{})
+
+func (s *txnManagerSuite) TestRunTxnManager(c *check.C) {
+	input := make(chan *Txn)
+	dmls := []*DML{
+		{Tp: UpdateDMLType},
+		{Tp: InsertDMLType},
+		{Tp: UpdateDMLType},
+	}
+	txn := &Txn{DMLs: dmls}
+	txnManager := newTxnManager(14, input)
+	output := txnManager.run()
+	// send 5 txns (size 3) to txnManager, the 5th txn should get blocked at cond.Wait()
+	for i := 0; i < 5; i++ {
+		select {
+		case input <- txn:
+		case <-time.After(50 * time.Microsecond):
+			c.Fatal("txnManager gets blocked while receiving txns")
+		}
+	}
+	c.Assert(output, check.HasLen, 4)
+	// Next txn should be blocked
+	select {
+	case input <- txn:
+		c.Fatal("txnManager doesn't block the txn when room is not enough")
+	case <-time.After(50 * time.Microsecond):
```

Comment:
```suggestion
	default:
```

Reply:
I think wait some time is better. Since `input` has no buffer if `txnManager` is not exactly [waiting to get item from `input`](https://github.com/pingcap/tidb-binlog/pull/737/files#diff-b39261d071d61502be586d1b85e6dfb5R667) it can be passed. But in fact txnManager may still be busy popping items or doing some other things when we check here.

Reply:
it's more acurrate to use `default` here, because  `the 5th txn should get blocked at cond.Wait()` or `after <- input and before cond.Wait()`